R/course.R

```diff
@@ -0,0 +1,29 @@
+download_zip <- function(url) {
+  dl <- curl::curl_fetch_memory(url)
+
+  httr::stop_for_status(dl$status_code)
+  stopifnot(
+    grepl("^https://dl.dropboxusercontent.com/content_link_zip/", dl$url) ||
+      grepl("^https://codeload.github.com", dl$url)
+  )
+
+  hh <- curl::parse_headers_list(dl$headers)
+  stopifnot(hh[["content-type"]] == "application/zip")
+  content_disposition <- hh[["content-disposition"]]
+  stopifnot(!is.null(content_disposition), nzchar(content_disposition))
```

Comment:
Too rigid? A filename could potentially also be generated from either the input url or `dl$url`.